clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -526,6 +519,67 @@ protected void coordinatorDead() {
     }
 
     /**
+     * Close the coordinator, waiting if needed to send LeaveGroup.
+     */
+    @Override
+    public void close() {
+        maybeLeaveGroup(true);
+    }
+
+    /**
+     * Leave the current group and reset local generation/memberId.
+     */
+    public void maybeLeaveGroup(boolean awaitResponse) {
+        if (!coordinatorUnknown() && generation > 0) {
+            // this is a minimal effort attempt to leave the group. we do not
+            // attempt any resending if the request fails or times out.
+            sendLeaveGroupRequest(awaitResponse);
+        }
+
+        this.generation = OffsetCommitRequest.DEFAULT_GENERATION_ID;
+        this.memberId = JoinGroupRequest.UNKNOWN_MEMBER_ID;
+        rejoinNeeded = true;
+    }
+
+    private void sendLeaveGroupRequest(boolean awaitResponse) {
+        LeaveGroupRequest request = new LeaveGroupRequest(groupId, memberId);
+        RequestFuture<Void> future = client.send(coordinator, ApiKeys.LEAVE_GROUP, request)
+                .compose(new LeaveGroupResponseHandler());
+
+        future.addListener(new RequestFutureListener<Void>() {
+            @Override
+            public void onSuccess(Void value) {}
+
+            @Override
+            public void onFailure(RuntimeException e) {
+                log.debug("LeaveGroup request failed with error", e);
```

Comment:
I feel INFO level is fine as in many cases people will keep alerting on warn / error logs.